### ISSUE TITLE
Bug: The selected data type from Entity or QueryBuilder is defferent from DB field type.

### ISSUE THREAD

**Issue by kai890707** [codeigniter4/CodeIgniter4#7259]
### PHP Version

8.1

### CodeIgniter4 Version

4.1.5

### CodeIgniter4 Installation Method

Git

### Which operating systems have you tested for this bug?

Linux

### Which server did you use?

cli

### Database

MariaDB 10.1

### What happened?

The data type selected is different from the DB field type.
For example, if you use a migrate file to define a field type as INTEGER or FLOAT, and then execute `php spark migrate` on the database to create a table, the data type returned using Query Builder will be different from the type initially set.

### Steps to Reproduce

1. Use Migration make a migrate file.
2. addField into mugrate file.
```php
          $this->forge->addField([
            'is_string'    => [
                'type'           => 'VARCHAR',
                'constraint'     => 30,
                'comment'        => "is_string"
            ],
            'is_integer'   => [
                'type'           => 'INT',
                'constraint'     => 30,
                'comment'        => "is_integer"
            ],
            'is_float'   => [
                'type'           => 'FLOAT',
                'comment'        => "is_float"
            ],
        ]);
        $this->forge->addKey('is_string', true);
        $this->forge->createTable('CastTable');
```
3. Use db query in Controller to get data
```php
$db = \Config\Database::connect();
$queryData = $db->table('CastTable')->get()->getResult();
var_dump($queryData);
```
4.Get data of the wrong type
```
array(1) {
  [0]=>
  object(stdClass)#57 (3) {
    ["is_string"]=>
    string(17) "Hi CodeIgniter4 !"
    ["is_integer"]=>
    string(3) "999"
    ["is_float"]=>
    string(5) "9.999"
  }
}
```

### Expected Output

```
array(1) {
  [0]=>
  object(stdClass)#57 (3) {
    ["is_string"]=>
    string(17) "Hi CodeIgniter4 !"
    ["is_integer"]=>
    int(999)
    ["is_float"]=>
    float(9.999)
  }
}
```

**Comment by kai890707** [comment]
I finished fixing the errors in this section and I can send a PR.

**Comment by kenjis** [comment]
This is not a bug in CodeIgniter.
It is the specification of PHP database module like `MySQLi`.

**Comment by kai890707** [comment]
So does this need to be adjusted by the user? Or can CodeIgniter 4 add this part of the fix?

**Comment by tangix** [comment]
> So does this need to be adjusted by the user? Or can CodeIgniter 4 add this part of the fix?

You could use CI's Model with Entity providing the casts needed.

**Comment by kenjis** [comment]
Closed by #7265
